astropy/io/fits/tests/test_image.py

```diff
@@ -1725,3 +1725,12 @@ def test_bzero_mishandled_info(tmpdir):
     hdu.writeto(filename, clobber=True)
     hdul = fits.open(filename)
     hdul.info()
+
+
+def test_image_write_readonly(tmpdir):
+    # Regression test to make sure that we can write out read-only arrays (#5512)
+    x = np.array([1,2,3])
+    x.setflags(write=False)
+    ghdu = fits.ImageHDU(data=x)
+    ghdu.add_datasum()
+    ghdu.writeto(tmpdir.join('test.fits').strpath)
```

Comment:
It could be useful to read the data and check that it was correctly written (with the good byte order). Also could you add the same test for `CompImageHDU` and `GroupsHDU` (I never used this one, but I guess it works similarly) ?